### ISSUE TITLE
tree-view.js and toggling visibility of local treeview

### ISSUE THREAD

**Issue by samifouad** [icetee/remote-ftp#652]
Hey there, first I want to say thank you for making this awesome plugin. I used Atom with a separate SFTP client previously and that way is so much more of a hassle. Keep up the great work.

The only thing I found to be some what of a minor problem is the local treeview doesn't seem to want to go away to expose just remote-ftp. If I want more space for my code and quickly toggle remote-ftp into view, once I toggle it away, the local treeview doesn't seem to want to go also.

After looking at the source code, I now understand remote-ftp is like a child node of the local treeview, so when it's launched, there the local treeview is, too. You seem to have made a setting option for this toggling scenario, but remote-ftp is not recognizing the setting.

From **line 242** of _blob/master/lib/views/tree-view.js_
`if (atom.config.get('Remote-FTP.hideLocalWhenDisplayed'))
  hideLocalTree();
else
  showLocalTree();`

I put this into my remote-ftp configuration:

> "hideLocalWhenDisplayed":true

But there was no change. Then it hit me, you're calling **atom.config.get()** so after going to **File -> Config** in Atom and adding `"Remote-FTP.hideLocalWhenDisplayed": true` to Atom's config.cson, what happens now is it toggles between which view is displayed, but the panel stays open.

But what I think it's supposed to do, and what I essentially want is for the entire panel to toggle between being hidden and remote-ftp. I'm trying to save myself having to use two different keyboard shortcuts because my brain doesn't seem to like doing Ctrl+Alt+o and Ctrl+\ in quick succession, even after dozens of times over.

I don't yet have any experience with Atom's plugin system, so I wish I could tinker with this problem myself and submit a pull request. Any help with this annoyance would be greatly appreciated. Cheers. :)

**Comment by TeLiXj** [comment]
I have the same problem since many versions ago. It's annoying :(
I only use local tree view for local projects, for this I always hidde this panel when use remote projects. But now when you open a project or toggle remote panel, the local panel always come back to appear. hideLocalWhenDisplayed isn't a solution because only toggle between local and remote panels as @samifouad said.
My temp solution is hide the local panel using my personal stylesheet. But it's tedious comment and uncomment this lines depending of project type.
```
//hide local panel
.tree-view-resizer {
    display:none;
}
```
I hope u can fix this because this package it's one of my favorites. Good job.

**Comment by jpxd** [comment]
If you don't want any tree view visible, can't you just toggle away the remote tree view (`Remote-FTP: Toggle`) and toggle away the local tree view (`Tree View: Toggle`)?
I'm using the command palette for this but there are also keyboard shortcuts.
<img width="645" alt="screenshot 2017-01-25 11 48 27" src="https://cloud.githubusercontent.com/assets/10384363/22287771/594ff704-e2f4-11e6-9a72-deb22f8dbb7b.png">

**Comment by jpxd** [comment]
@samifouad Do i understand correctly, that you want a keyboard shortcut, which toggles the Remote-FTP treeview and Atoms treeview *together*?

**Comment by TeLiXj** [comment]
@jpxd I think not. The problem is if you only use remote ftp for edit directly in the server u don't need local treeview, but always appears when you toggle remote treeview. IMHO it's better fix this behaviour instead of add a especial keyboard shorcut to toggle off both treeviews

**Comment by jpxd** [comment]
So you want: Atoms treeview is hidden and remote-ftp one is visible and you toggle the Remote-ftp treeview -> no treeview is visible. correct?

**Comment by samifouad** [comment]
Yeah, that's exactly right.

You can easily toggle local treeview with a shortcut. You cannot easily toggle visibility for remote-ftp without having to deal with an extra panel that has another shortcut associated with it to close.

eg. (on Windows) Ctrl+Alt+O will open remote-ftp and treeview. To close what you just opened, you then have to use two separate shortcuts. Ctrl+Alt+O again AND Ctrl+\

**Comment by samifouad** [comment]
I just wanted to post that I think I was able to figure out a fix.

I figured out how to edit the package source code and I commented out line 240 of blob/master/lib/views/tree-view.js

After restarting Atom, remote and local treeview works as I think it should. Ctrl+Alt+O opens ONLY remote treeview. Ctrl+\ opens ONLY local treeview.

I'll see if I can submit a pull request to help fix this for more than just my computer.
